### PR TITLE
fix(sync/drain): default FullCorpus to false when request body is absent

### PR DESCRIPTION
**Pull request by bsvalues**
## Summary

- `DoctrineDrainController.NormalizeRequest` previously defaulted `FullCorpus ?? true` — a missing request body silently triggered full-corpus imports
- Changed default to `?? false`: bodyless calls now resolve to bounded mode (per-lane TopN safe default of 200)
- `NormalizeRequest` changed `private → internal` to enable direct unit testing via existing `InternalsVisibleTo` grant
- 5 new unit tests added covering: null-body bounded default, bounded body pass-through, explicit full-corpus body, operator-name default, working-year default
- 3414/3414 unit tests pass — no regressions

**Root cause (from PR #1050):** SCALE-001A and SCALE-001B drain calls used `?topN=500` query-string with no body. The endpoint is `[FromBody]` only — query-string TopN is not bound. Null body → `FullCorpus=true` → full corpus.

**Prior evidence (FIX3–FIX7B) unaffected:** All prior controlled-slice drains sent correct JSON bodies with `"FullCorpus":false,"TopN":100` and remain valid.

**Explicit full-corpus still works:** `{"FullCorpus":true}` in the body triggers full corpus as before.

## Test plan

- [x] 5 new `DoctrineDrainNormalizeRequestTests` — all PASS
- [x] Full `TerraFusion.Unit.Tests` suite — 3414/3414 PASS
- [x] Pre-push gate (vitest 164/164 + security scan) — PASS
- [ ] CI Backend Gate
- [ ] CI Frontend Gate

## Evidence

`docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md`

**WO:** WO-DATA-004B-SCALE-001Y
**Prereq:** PR #1050 (SCALE-001X investigation)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Adjust Doctrine drain request normalization to use a safe bounded default when no request body is provided and add coverage and documentation for the new behavior.

Bug Fixes:
- Prevent bodyless Doctrine drain calls from defaulting to full-corpus imports by defaulting FullCorpus to false instead of true.

Enhancements:
- Expose NormalizeRequest as internal to allow direct unit testing via existing InternalsVisibleTo configuration.

Documentation:
- Add SCALE-001Y evidence document describing the safe-default patch, affected endpoints, behavior matrix, and correct drain usage.

Tests:
- Add DoctrineDrainNormalizeRequest unit test suite covering null-body defaults, bounded and full-corpus behaviors, and operator/year defaults.

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated default behavior for data drainage requests: bulk operations now default to selective retrieval instead of full-corpus extraction when the parameter is omitted, reducing unintended data volume.

* **Documentation**
  * Updated API documentation to reflect the new default parameter behavior and provide guidance for proper drain request configuration.

* **Tests**
  * Added unit tests validating drain request normalization and default parameter handling across various input scenarios.

### CODE REVIEW

File: docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+**Accessor change:** `NormalizeRequest` was also changed from `private static` to `internal static`
+to enable direct unit testing via the existing `InternalsVisibleTo` grant to `TerraFusion.Unit.Tests`.
+
+---
+
+## 3. Behavior Matrix After Patch
+
+| Call pattern | `request` | `FullCorpus` resolved | `seedTopN` | Effect |
+|---|---|---|---|---|
+| No body (curl without -d) | `null` | `false` ← patched | `200` (per-lane default) | Bounded 200-row slice |
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: backend/tests/TerraFusion.Unit.Tests/Sync/Doctrine/DoctrineDrainNormalizeRequestTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    [Fact]
+    public void NullBody_operator_name_defaults_to_lane_scoped_name()
+    {
+        var (operatorName, _, _, _) = DoctrineDrainController.NormalizeRequest(null, "owner-wsdor");
+
+        operatorName.Should().Be("doctrine-drain-owner-wsdor");
+    }
+
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply backend/tests/TerraFusion.Unit.Tests/Sync/Doctrine/DoctrineDrainNormalizeRequestTests.cs:63]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: backend/src/TerraFusion.API/Controllers/DoctrineDrainController.cs [hunk 3]
```diff
@@ -1637,14 +1637,14 @@ private static (string? cs, IActionResult? error) ResolveConnectionString(IConfi
...
+    internal static (string OperatorName, short WorkingYear, bool FullCorpus, int? TopN)
         NormalizeRequest(DoctrineDrainRequest? request, string laneName)
     {
         var operatorName = string.IsNullOrWhiteSpace(request?.OperatorName)
             ? $"doctrine-drain-{laneName}"
             : request!.OperatorName!.Trim();
         var workingYear = (short)(request?.WorkingYear ?? 2026);
-        var fullCorpus = request?.FullCorpus ?? true;
+        var fullCorpus = request?.FullCorpus ?? false;
         var topN = request?.TopN;
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply backend/src/TerraFusion.API/Controllers/DoctrineDrainController.cs:1648]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+triggered full-corpus mode on every bodyless call. SCALE-001A completed a near-full Benton
+parcel corpus import (~93%, 83,326 rows). SCALE-001B was killed after ~1 hour with 0 rows
+committed. See `PACS_SYNC_SCALE_001X_TOPN_PROPAGATION_FAILURE.md` for the full investigation.
+
+---
+
+## 2. The Fix
+
+**File:** `backend/src/TerraFusion.API/Controllers/DoctrineDrainController.cs`
+**Location:** `NormalizeRequest` private method, line 1647
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+## 8. Correct Drain Call Format (post-patch)
+
+Always send a JSON body. The endpoint binding is `[FromBody]` only — there is no
+query-string TopN binding.
+
+```bash
+# Bounded drain (recommended for scale proof):
+curl -X POST "http://localhost:5046/api/sync/doctrine/drain/parcel" \
+  -H "Content-Type: application/json" \
+  -d '{"OperatorName":"scale-001a-parcel","WorkingYear":2026,"FullCorpus":false,"TopN":500}'
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+# Full corpus (explicit opt-in):
+curl -X POST "http://localhost:5046/api/sync/doctrine/drain/parcel" \
+  -H "Content-Type: application/json" \
+  -d '{"OperatorName":"full-corpus-run","WorkingYear":2026,"FullCorpus":true}'
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.

File: docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+The following call form is NOW SAFE (no body → bounded at 200 rows, not full corpus):
+```bash
+curl -X POST "http://localhost:5046/api/sync/doctrine/drain/parcel"
```
[the remark this answers is not in the document]
Comment (bsvalues) [reply docs/data/PACS_SYNC_SCALE_001Y_SAFE_DEFAULT_PATCH.md]: Fixed in 983bf010: updated XML doc, added Theory test (null/empty/whitespace OperatorName), corrected 'private'→'internal' in doc, corrected sales per-lane default to 500, replaced hardcoded port with ${TF_API_PORT:-5046} in all three examples.
